test-resources-plugin/src/main/java/io/micronaut/gradle/testresources/MicronautTestResourcesConsumerPlugin.java

```diff
@@ -39,23 +37,36 @@ public class MicronautTestResourcesConsumerPlugin implements Plugin<Project> {
     public void apply(Project project) {
         PluginManager pluginManager = project.getPluginManager();
         pluginManager.apply(MicronautBasePlugin.class);
-        TestResourcesConsumerConfiguration testResourcesExtension = createTestResourcesExtension(project);
-        pluginManager.withPlugin("io.micronaut.component", unused -> {
-            project.afterEvaluate(p -> p.getConfigurations().all(cnf -> configureDependencies(p, testResourcesExtension, project.getDependencies(), cnf)));
+        Configuration testResourcesConfiguration = createTestResourcesExtension(project);
+        pluginManager.withPlugin("io.micronaut.component", unused ->
+                project.afterEvaluate(p -> p.getConfigurations().all(cnf -> configureDependencies(testResourcesConfiguration, cnf)))
```

Comment:
Is this afterEvaluate still needed?

Reply:
I will double check. A `configureEach` should work in theory. In the other plugin I had to defer configuration.

Reply:
I thought Configurations were never lazy, so .all should be fine

Reply:
They are not, but if your `.all` action executes synchronously if the configuration exists, and asynchronously if not. This caused some weird bugs.